### ISSUE TITLE
[Tracking]: Evaluation Integrations 

### ISSUE THREAD

**Issue by jon-chuang** [run-llama/llama_index#6824]
### Feature Description

- [ ] Retrieval: https://github.com/jerryjliu/llama_index/issues/6554
- [ ] RAG / Knowledge-Intensive QA: https://github.com/jerryjliu/llama_index/issues/6919
- [ ] MMLU?
- [ ] https://github.com/openai/evals?

TODO: investigate more RAG-specific benchmarks rather than retrieval-only or generation-only.

### References / Areas of Exploration
1. [Academic work on FLARE](https://arxiv.org/abs/2305.06983) - RAG + reranking. Can get some inspiration from their evaluation methodology
2. [Wizard of Wikipedia - Knowledge Augmented Dialogue](https://parl.ai/projects/wizard_of_wikipedia/). Not sure if there is any evaluation.
3. [Evaluation of Dataset Selection for Pre-Training and Fine-Tuning Transformer Language Models for Clinical Question Answering](https://aclanthology.org/2020.lrec-1.679.pdf)
   - [CLiCR](https://paperswithcode.com/dataset/clicr)
4. [Streaming QA](https://github.com/deepmind/streamingqa).
   - The nice thing about this dataset is that one can potentially segment out the dataset. But I'm not sure if this is possible to do (i.e. from `t=t_1 to t=t_2`) v.s. an aggregate over time from `t=0`. Otherwise the dataset is too massive (110GB compressed).  Even the index is massive (400MB+). See issue: https://github.com/deepmind/streamingqa/issues/2
   - Segment: https://data.statmt.org/news-crawl/en/news.2007.en.shuffled.deduped.gz ~ 170M. Still pretty large but probably some kind of segmenting can be done in realtime using a BTree / something like Polars DB.

### Target Domains
1. Finance: [FinQA](https://github.com/czyssrs/FinQA) - [FinTabNet Corpus](https://developer.ibm.com/exchanges/data/all/fintabnet/). 
2. Codebase?? e.g. [codequeries](https://github.com/thepurpleowl/codequeries-benchmark), [CodeQA](https://paperswithcode.com/dataset/codeqa)

TODO:
1. Compile all of the dataset sizes (we should prefer fine-tuning datasets with out-of-domain corpora rather than large and generic corpora which are likely representative of the LLM training datasets)
